Scenario 01-Appt Scheduling Calendar View

Athale08 opened the pull request.

Athale08 committed:
Scenario 01-Appt Scheduling Calendar View

Athale08 committed:
Scenario 01-Appt Scheduling Calendar View

jdhage662 commented on src/main/java/appPages/BaseAction.java:
```diff
@@ -1,28 +1,88 @@
+
+        wait.until(ExpectedConditions.visibilityOfElementLocated(by));
+    }
+    //sum(int a, int b) ===> sum(2,3)
```
Code should not contain the commented line

Athale08 replied:
Done.

jdhage662 commented on src/main/java/appPages/BaseAction.java:
```diff
@@ -1,28 +1,88 @@
+        wait.until(ExpectedConditions.visibilityOfElementLocated(by));
+    }
+    //sum(int a, int b) ===> sum(2,3)
+    //sum(int a , int b, int c) ==> sum(2,3,4)
```
same as above

Athale08 replied:
Done.

jdhage662 commented on src/main/java/appPages/BaseAction.java:
```diff
@@ -1,28 +1,88 @@
+    }
+    //sum(int a, int b) ===> sum(2,3)
+    //sum(int a , int b, int c) ==> sum(2,3,4)
+    // sum(int a , int b, int c , int d) ==> sum(2,3,4,5)
```
SAME AS ABOVE

Athale08 replied:
Done.

jdhage662 commented on src/main/java/appPages/BaseAction.java:
```diff
@@ -1,28 +1,88 @@
+    //sum(int a , int b, int c) ==> sum(2,3,4)
+    // sum(int a , int b, int c , int d) ==> sum(2,3,4,5)
+
+    public void clickElement(By by) {
```
instead of passing the locator technique pass the element

Athale08 replied:
Done.

jdhage662 commented on src/main/java/appPages/BaseAction.java:
```diff
@@ -33,8 +93,153 @@ public String getCurrentSystemDate(int noOfhourse) {
+        }
+    }
+    //Verifying element getText for dom elements
+    public void verifyElementContainsText(By by, String expectedResult){
```
pass the element instead of locator

Athale08 replied:
Done.

jdhage662 commented on src/test/java/stepDefinitions/AppointmentSchedulingCalendarViewStepDef.java:
```diff
@@ -0,0 +1,51 @@
+
+       manageAppointmentsPage.verifyCurrentWeeklyDate();
+    }
+
```
remove the spaces

Athale08 replied:
Done.

jdhage662 commented on src/test/java/stepDefinitions/LoginStepDefinition.java:
```diff
@@ -31,6 +30,4 @@ public void user_should_be_able_to_login_to_the_app() {
 
 
 
-
```
remove spaces

Athale08 replied:
Done.

jdhage662 commented on src/test/resources/featureFiles/AppointmentSchedulingCalenderViewTest.feature:
```diff
@@ -0,0 +1,15 @@
+@AptTest01
+ Scenario: Display appointments for the current week in Calendar view
+    Given on the Home page the user clicks on the Appointment Scheduling button
+    And user navigates to the "<Manage Appointments>" page
```
remove the <> as this is used when you will use the scenario outline with example table

Athale08 replied:
Done.

Athale08 committed:
Scenario 01-Appt Scheduling Calendar View:
1. Extra spaces removed
2. Comments removed, few are left for information purposes.
3.The step definitions now directly call the existing methods in ManageAppointmentsPage that interact with the WebElement instances, like clickOnAppointmentSchedulingButton(), clickOnAppointmentsListButton(), and clickOnWeekButton().
4. BaseAction Page-Each button is now declared as a WebElement instead of a By locator.

Athale08 committed:
Scenario 01-Appt Scheduling Calendar View:
1. Extra spaces removed
2. Comments removed, few are left for information purposes.
3.The step definitions now directly call the existing methods in ManageAppointmentsPage that interact with the WebElement instances, like clickOnAppointmentSchedulingButton(), clickOnAppointmentsListButton(), and clickOnWeekButton().
4. BaseAction Page-Each button is now declared as a WebElement instead of a By locator.

Athale08 committed:
Scenario 01-Appt Scheduling Calendar View:
1. Extra spaces removed
2. Comments removed, few are left for information purposes.
3.The step definitions now directly call the existing methods in ManageAppointmentsPage that interact with the WebElement instances, like clickOnAppointmentSchedulingButton(), clickOnAppointmentsListButton(), and clickOnWeekButton().
4. BaseAction Page-Each button is now declared as a WebElement instead of a By locator.

Athale08 committed:
Merge remote-tracking branch 'origin/feature01' into feature01

# Conflicts:
#	src/main/java/appPages/BaseAction.java
#	src/main/java/appPages/ManageAppointmentsPage.java
#	src/main/java/drivers/DriverManager.java
#	src/test/java/stepDefinitions/AppointmentSchedulingCalendarViewStepDef.java
#	src/test/resources/featureFiles/AppointmentSchedulingCalenderViewTest.feature

Athale08 committed:
Scenario 01-Appt Scheduling Calendar View:Refactor01
1. Extra spaces removed
2. Comments removed, few are left for information purposes.
3.The step definitions now directly call the existing methods in ManageAppointmentsPage that interact with the WebElement instances, like clickOnAppointmentSchedulingButton(), clickOnAppointmentsListButton(), and clickOnWeekButton().
4. BaseAction Page-Each button is now declared as a WebElement instead of a By locator.

Athale08 committed:
Scenario 02- Home Page Button user flow test

Athale08 commented:
@jdhage662 - merging is blocked. Could you pls approve this?

Athale08 commented:
@jdhage662 
Merging is blocked. Could you please approve this so that I can merge the pull request. Thank you.
